src/meta/processors/job/MetaJobExecutor.cpp

```diff
@@ -79,15 +79,21 @@ ErrOrHosts MetaJobExecutor::getTargetHost(GraphSpaceID spaceId) {
   }
 
   // use vector instead of set because this can convient for next step
+  std::unordered_map<HostAddr, std::vector<PartitionID>> hostAndPart;
   std::vector<std::pair<HostAddr, std::vector<PartitionID>>> hosts;
   while (iter->valid()) {
+    auto part = MetaServiceUtils::parsePartKeyPartId(iter->key());
     auto targets = MetaServiceUtils::parsePartVal(iter->val());
     for (auto& target : targets) {
+      hostAndPart[target].emplace_back(part);
       std::vector<PartitionID> parts;
       hosts.emplace_back(std::make_pair(std::move(target), std::move(parts)));
```

Comment:
If you use map to collect all parts of each host, it's not necessary to push them into the hosts container here and dedup them by std::unique in following steps.

If you need to dedup the partIDs, you could replace the std::vector with std::unordered_set in unordered_map.

Reply:
Actually I need a vector<pair> to collect the host and partition mapping

Reply:
I see you have copied the contents of map to the vector in the code below, so need not to push them to hosts here.